aten/src/ATen/native/mps/operations/Indexing.mm

```diff
@@ -252,31 +252,42 @@ Tensor nonzero_fallback(const Tensor& self) {
     MPSGraphTensor* inputTensor_ = nil;
     MPSGraphTensor* outputTensor_ = nil;
     MPSGraphTensor* scatterDataTensor_ = nil;
+    MPSGraphTensor* countNonzeroTensor_ = nil;
   };
 
-  int64_t total_nonzero = at::count_nonzero(self).item<int64_t>();
-  at::native::resize_output(out_, {total_nonzero, nDim});
-  if (out_.numel() ==  0) {
-    return out_;
-  }
-
-  bool contiguous_output = (out_.is_contiguous() && !out_.is_view());
-  Tensor out = out_;
-  if (!contiguous_output) {
-    out = at::native::empty_mps(
-           out_.sizes(),
+  // int64_t total_nonzero = at::count_nonzero(self).item<int64_t>();
```

Comment:
Do we want to upstream this block of commented code? If not, I suggest you remove it and keep it locally.

Reply:
Thanks - yes, this needs to be removed (part of old code). I'll update